oktapam/data_source_ad_user_sync_task_settings_list.go

```diff
@@ -0,0 +1,135 @@
+package oktapam
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/typed_strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/okta/terraform-provider-oktapam/oktapam/client"
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/attributes"
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/descriptions"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceADUserSyncTaskSettingsList() *schema.Resource {
+	return &schema.Resource{
+		Description: descriptions.SourceADUserSyncTaskSettingsList,
+		ReadContext: dataSourceADUserSyncTaskSettingsListRead,
+		Schema: map[string]*schema.Schema{
+			// Query parameter values
+			attributes.ADUserSyncTaskSettingsStatus: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: descriptions.FilterADUserSyncTaskSettingsStatus,
+				ValidateFunc: validation.StringInSlice(
+					[]string{
+						typed_strings.TaskSettingsStatusActive.String(),
+						typed_strings.TaskSettingsStatusInActive.String(),
+					}, false),
+			},
+			attributes.ADConnectionID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: descriptions.FilterConnectionID,
+			},
+			// Return value
+			attributes.ADUserSyncTaskSettingsList: {
```

Comment:
I missed this part in my initial review. For plural data sources, we are just returning IDs only instead of the whole resource object. This standard is followed in most of the tf providers. Check datasource data_source_projects.go for implementation. I know not all of the datasources(gateways, ad_connections) in the oktapam provider following this standard because those were created before. I will create user stories to fix those too so that we have consistent code.

Reply:
ok will follow that pattern